[ENH] JohnsonQPDRegressor for quantile to probability conversion

RakshithaKowlikar opened the pull request:
This PR adds JohnsonQPDRegressor, a wrapper that transforms any sklearn-style quantile regressor into a probabilistic regressor using the Johnson Quantile-Parameterized Distribution (QPD).

Fixes #377

- Fits three quantile models at (α, 0.5, 1−α)
- Constructs a QPD_Johnson distribution from predicted quantiles
- Enforces quantile ordering and minimum spread for stability
- Optional bounded support via (lower, upper)
- Compatible with any sklearn regressor that has a `quantile` parameter

### Modified Files
* `skpro/regression/compose/_johnson_quantile.py` - New implementation
* `skpro/regression/compose/__init__.py` - Added import
* `docs/source/api_reference/regression.rst` - Added to API documentation

fkiraly reviewed (commented):
Very nice - I've started the tests.

fkiraly reviewed (changes requested):
Quick question: why are you making changes to the qpd distributions? If these need changes, then maybe open a separate PR and explain what you are doing?

RakshithaKowlikar commented:
Changes to `qpd.py`:
- sample() method was added to fix `TypeError: object of type 'NoneType' has no len()`. Because QPD_Johnson was created with scalar parameters (no index/columns specified), self.index was None.
- theta was clamped to prevent arccosh from receiving invalid inputs when symmetric distribution occurs.
- wrap in `np.errstate` to suppress RuntimeWarnings during intermediate calculations because intermediate calculations can produce inf, NaN, or divide-by-zero even though the final result might be valid
- kappa calculation safety to check for zero before dividing.

Changes in `_johnson_quantile.py`:
- changed the spread calculation so that it works across different data scales, works when quantiles are too close together now
- added _predict_var because when ppf(scalar) is called on a distribution with shape (13, 1), it tries to return shape (13, 13), because ppf is in the base class, shared across all distributions. Fixing it there could break other distributions

@fkiraly should i still open a pr for the changes in `qpd.py`?

RakshithaKowlikar commented:
Also the errors seem to be linked with `GAMRegressor` where `get_test_params()` is returning a single dict
